pkg/cloudprovider/aws/cloudprovider.go

```diff
@@ -125,7 +125,7 @@ func (c *CloudProvider) create(ctx context.Context, constraints *v1alpha4.Constr
 	// Partial fulfillment will be logged
 	nodes, err := c.instanceProvider.Create(ctx, vendorConstraints, instanceTypes, quantity)
 	if err != nil {
-		return fmt.Errorf("launching %d instance(s), %w", quantity, err)
+		return fmt.Errorf("launching %d instance(s) with constraints %+v and instance types %+v, %w", quantity, vendorConstraints, instanceTypes, err)
```

Comment:
We try to avoid explosively complex log lines, at least for commonly logged things. Now that I see this, maybe we should revise this idea that our Info/Debug should avoid data dumps like this, but it's fine for Error().

Check out the pretty.Concise() helper to jsonify these. I used to have pretty.Verbose() that included newlines, but this somehow got removed when I was on vacation :(. It's here if you need a reference: https://github.com/ellistarn/slang/blob/main/pkg/pretty/pretty.go